### PR TITLE
feat(VNavigationDrawer): add smooth touch dragging

### LINKED ISSUE #3262: [Feature Request] Smooth Slide Navigation Drawer (see quasar)
### New Functionality
On mobile (cordova) I want the navigation drawer to feel more native. See quasar (http://quasar-framework.org/quasar-play/android/index.html#/showcase/layout/play-with-layout) in mobile emulation mode. As you drag your finger from either edge the navigation drawer follows your finger. Whereas, with vuetify, it pops out when the gesture is finished. 

This is borderline necessary in order for me to pursue vuetify as my component framework in a cordova application. 

### Improvements
NavigationDrawer gesture/animation

### Bugs or Edge Cases it Helps Avoid
n/a

### PR DESCRIPTION
**Pull request by KaelWD**
## Description

closes #3262

## Markup:

<details>

```vue
<template>
  <v-app>
    <v-container>
      <v-navigation-drawer app>
        Drawer
      </v-navigation-drawer>
      <v-main>
        Content
      </v-main>
    </v-container>
  </v-app>
</template>
```
</details>

### CODE REVIEW

File: packages/vuetify/src/components/VNavigationDrawer/touch.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
+import { useVelocity } from '@/composables/touch'
```
Comment (johnleider) [inline packages/vuetify/src/components/VNavigationDrawer/touch.ts:1]: should this be a reusable composable?

File: packages/vuetify/src/composables/touch.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import { CircularBuffer } from '@/util'
```
[the remark this answers is not in the document]
Comment (KaelWD) [reply packages/vuetify/src/composables/touch.ts:1]: I was going to add more touch stuff to here (and might still to support global drawers). useVelocity is currently specific to touch velocity.

File: packages/vuetify/src/components/VNavigationDrawer/touch.ts (hunk 1 above)
Comment (KaelWD) [reply packages/vuetify/src/components/VNavigationDrawer/touch.ts:1]: Everything in here is specific to drawers

File: packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass [hunk 3]
```diff
@@ -77,3 +80,14 @@
     height: $navigation-drawer-img-height
     object-fit: $navigation-drawer-img-object-fit
     width: $navigation-drawer-img-width
+
+.v-navigation-drawer__scrim
+  position: fixed
```
[the remark this answers is not in the document]
Comment (KaelWD) [reply packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass:85]: The only properties it'd reuse from that are top and right

**Review by johnleider — APPROVED** [review]

**Pull request merged**